lib/block_view_test.go

```diff
@@ -18895,6 +19001,500 @@ func TestNFTPreviousOwnersCantAcceptBids(t *testing.T) {
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m3Pub,
+			m3Priv,
+			post1Hash,
+			1,    /*SerialNumber*/
+			20, /*BidAmountNanos*/
+		)
+
+		// There are three bids now.
+		bidEntries = DBGetNFTBidEntries(db, post1Hash, 1)
+		require.Equal(3, len(bidEntries))
+
+		nftEntry := DBGetNFTEntryByPostHashSerialNumber(db, post1Hash, 1)
+
+		// m2 is still the owner of the NFT since this is not a buy now.
+		m2PKID := DBGetPKIDEntryForPublicKey(db, m2PkBytes)
+		require.Equal(nftEntry.OwnerPKID, m2PKID.PKID)
+	}
+
+	// Have m2 accept bid from m3
+	{
+		bidAmountNanos := uint64(20)
+		// Creator Balance before.
+		m0BalBefore := _getBalance(t, chain, nil, m0Pub)
+		require.Equal(uint64(1023), m0BalBefore)
+
+		// Bidder Balance before.
+		m3BalBefore := _getBalance(t, chain, nil, m3Pub)
+		require.Equal(uint64(999), m3BalBefore)
+
+		// Seller Balance before.
+		m2BalBefore := _getBalance(t, chain, nil, m2Pub)
+		require.Equal(uint64(848), m2BalBefore)
+
+		// DESO locked before royalties.
+		m0DeSoLockedBefore, _ := _getCreatorCoinInfo(t, db, params, m0Pub)
+
+		// M2 accepts M3's bid.
+		_acceptNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m2Pub,
+			m2Priv,
+			post1Hash,
+			1, /*SerialNumber*/
+			m3Pub,
+			bidAmountNanos, /*BidAmountNanos*/
+			"", /*UnlockableText*/
+		)
+		// All outstanding bids on serial #1 are cancelled.
+		bidEntries := DBGetNFTBidEntries(db, post1Hash, 1)
+		require.Equal(0, len(bidEntries))
+
+		nftEntry := DBGetNFTEntryByPostHashSerialNumber(db, post1Hash, 1)
+
+		// m3 is now the owner of the NFT.
+		m3PKID := DBGetPKIDEntryForPublicKey(db, m3PkBytes)
+		require.Equal(nftEntry.OwnerPKID, m3PKID.PKID)
+
+		// Creator Balance after. M0's balance should increase by the creator royalties (2)
+		m0BalAfter := _getBalance(t, testMeta.chain, nil, m0Pub)
+		require.Equal(uint64(1025), m0BalAfter)
+		require.Equal(m0BalAfter, m0BalBefore + 2)
+
+		// Bidder Balance after. M1's balance should decrease by the bid amount (20)
+		m3BalAfter := _getBalance(t, testMeta.chain, nil, m3Pub)
+		require.Equal(uint64(979), m3BalAfter)
+		require.Equal(m3BalAfter, m3BalBefore - 20)
+
+		// Seller Balance after. m2's balance should increase by the bid amount (20) less creator royalties (2), coin royalties (4) and the transaction fee (2).
+		m2BalAfter := _getBalance(t, testMeta.chain, nil, m2Pub)
+		require.Equal(uint64(860), m2BalAfter)
+		require.Equal(m2BalAfter, m2BalBefore + 20 - 4 - 2 - 2)
+
+		// Make sure royalties to creator and to coin are paid out correctly.
+		expectedCreatorRoyalty := bidAmountNanos / 10
+		require.Equal(uint64(2), expectedCreatorRoyalty)
+		expectedCoinRoyalty := 2 * bidAmountNanos / 10
+		require.Equal(uint64(4), expectedCoinRoyalty)
+		desoLocked, _ := _getCreatorCoinInfo(t, db, params, m0Pub)
+		require.Equal(m0DeSoLockedBefore+expectedCoinRoyalty, desoLocked)
+	}
+
+	// Roll all successful txns through connect and disconnect loops to make sure nothing breaks.
```

Comment:
I'd add something with a transfer + buy now if you can. The IsPending is an interesting edge case.

Other things to consider that you might have done that I might have missed:
* I think you might have done this and I missed it, but bidding with an amount that's below the min for IsBuyNow=true is interesting.
* Also bidding zero when you have IsBuyNow is interesting.